prover/src/constraints/evaluator/logup_gkr.rs

```diff
@@ -293,3 +342,23 @@ impl<E: FieldElement> TransitionDivisorEvaluator<E> {
             - E::BaseField::ONE
     }
 }
+
+/// Computes the evaluations of the s-column divisor.
+///
+/// The divisor for the s-column is $X^n - 1$ where $n$ is the trace length. This means that
+/// we need only compute `ce_blowup` many values and thus only that many exponentiations.
+fn compute_s_col_divisor<E: FieldElement>(
+    ce_domain_size: usize,
+    domain: &StarkDomain<E::BaseField>,
+    trace_length: usize,
+) -> Vec<E::BaseField> {
+    let degree = trace_length as u32;
+    let mut result = Vec::with_capacity(ce_domain_size);
+
+    for row in 0..domain.trace_to_ce_blowup() {
+        let x = domain.get_ce_x_at(row).exp(degree.into()) - E::BaseField::ONE;
+
+        result.push(x);
+    }
+    batch_inversion(&result)
+}
```

Comment:
Something I'm not following in this function: why do we allocate the `result` vector to have `ce_domain_size` capacity, but the fill only the first `trace_to_ce_blowup` values of it?

Also, why do we pass `ce_domain_size` separately into this function? Can we not get it from the `domain` struct?

Reply:
Good catch! Sorry about the confusion, I think this was a leftover from before I optimized it to use only `domain.trace_to_ce_blowup()` entries.
Updated now the capacity and removed the `ce_domain_size` parameter.